.github/workflows/main.yml

```diff
@@ -92,67 +85,68 @@ jobs:
         run: |
           echo "::set-output name=${{ matrix.os_short }}::https://storage.googleapis.com/taqueria-artifacts/${{ steps.upload-file.outputs.uploaded }}"
 
-  build-and-publish-npm-packages:
-    runs-on: ubuntu-latest
-    permissions:
-      packages: write
-      contents: read
-    strategy:
-      matrix:
-        include:
-          # - dir: taqueria-plugin-flextesa
-          - dir: taqueria-plugin-ligo
-          - dir: taqueria-plugin-smartpy
-          - dir: taqueria-plugin-taquito
-          # - dir: taqueria-plugin-teztnets
-          - dir: taqueria-sdk
-    defaults:
-      run:
-        working-directory: ${{ matrix.dir }}
-    env: 
-      NODE_AUTH_TOKEN: ${{ secrets.NPMJS_PAT }}
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
-        with: 
-          registry-url: ${{ env.NPM_REGISTRY }}
-          scope: "${{ env.SCOPE }}"
-
-      - name: Check tag suffix
-        id: check-tag-suffix
-        if: startsWith(github.ref, 'refs/tags/v') && (contains(github.ref, '-rc') || contains(github.ref, '-beta') || contains(github.ref, '-test'))
-        run: |
-          echo "::set-output name=npm-tag::next"
-
-      - name: Install any dependencies
-        run: npm ci
+  publish-protocol-to-npm:
+    needs: build-binaries
+    uses: ecadlabs/taqueria/.github/workflows/npm-publish.yml@${{github.sha}}
+    with:
+      dir: taqueria-protocol
+    secrets:
+      NPM_TOKEN: ${{ secrets.NPMJS_PAT }}
+
+  publish-sdk-to-npm:
+    needs: publish-protocol-to-npm
+    uses: ecadlabs/taqueria/.github/workflows/npm-publish.yml@${{github.sha}}
+    with:
+      dir: taqueria-sdk
+    secrets:
+      NPM_TOKEN: ${{ secrets.NPMJS_PAT }}
+
+  publish-ligo-plugin-to-npm:
+    needs: publish-sdk-to-npm
+    uses: ecadlabs/taqueria/.github/workflows/npm-publish.yml@alex-npm-pre-build-ordering
+    with:
+      dir: taqueria-plugin-ligo
+    secrets:
+      NPM_TOKEN: ${{ secrets.NPMJS_PAT }}
+  
+  publish-smartpy-plugin-to-npm:
+    needs: publish-sdk-to-npm
+    uses: ecadlabs/taqueria/.github/workflows/npm-publish.yml@alex-npm-pre-build-ordering
+    with:
+      dir: taqueria-plugin-smartpy
+    secrets:
+      NPM_TOKEN: ${{ secrets.NPMJS_PAT }}
+
+  publish-taquito-plugin-to-npm:
+    needs: publish-sdk-to-npm
+    uses: ecadlabs/taqueria/.github/workflows/npm-publish.yml@alex-npm-pre-build-ordering
+    with:
+      dir: taqueria-plugin-taquito
+    secrets:
+      NPM_TOKEN: ${{ secrets.NPMJS_PAT }}
+
+  publish-flextesa-plugin-to-npm:
+    needs:
```

Comment:
@alexzbusko great work thus far! 

Small thing - the flextesa plugin doesn't need these plugins to be built first:
- publish-ligo-plugin-to-npm
- publish-smartpy-plugin-to-npm
- publish-taquito-plugin-to-npm